### PR TITLE
[FLINK-3296] Remove 'flushing' behavior of the OutputFormat in DataStream API

### PR DESCRIPTION
**Pull request by rmetzger**
I removed the `FileSinkFunctionByMillis` and removed all the `millis` arguments on the writing functions.

The whole "buffering" and "flushing" functionality was broken: Elements were kept in an ArrayList and send to the OutputFormat on "flush()". However, the flush was not really called periodically. It was only checked when new records arrived. So when a stream is not having elements for a certain time, the last few elements would just stay in the list until new elements arrive again.

### CODE REVIEW

**Comment by mbalassi** [comment]
With the `RollingHDFSSink` this functionality is not needed any more and as you suggested apparently was a misleading implmentation anyway. I like your not to the docs. :+1: 
Could you add `[api-breaking]` to the commit msg though please, when merging?

File: docs/apis/streaming/index.md [hunk 1]
```diff
@@ -1768,6 +1768,11 @@ greater than 1, the output will also be prepended with the identifier of the tas
 </div>
 </div>
 
+Note that the `write*()` methods on `DataStream` are mainly intended for debugging purposes.
+They are not participating in Flink's checkpointing (no fault tolerance guarantees). The
```
Comment (StephanEwen) [inline docs/apis/streaming/index.md]: May be worth adding that this means usually "at-least-once" , but may also mean data loss in cases where the output formats buffer data and do not immediately persist it.

**Comment by StephanEwen** [comment]
Code looks good. I would like to update the docs to include a bit more info (like in the inline comment) and at least refer to the `addSink(...)` method, for fault tolerant sinks.

**Comment by tillrohrmann** [comment]
Changes look good to me

**Comment by rmetzger** [comment]
Thank you for the review. I've addressed the comments and rebased the change.

Once travis has passed, I'll merge it!

**Comment by aljoscha** [comment]
I think the `writeOutputFormat` method name could be misleading. To me it implies writing the `OutputFormat` not writing something by using the `OutputFormat`.

**Comment by rmetzger** [comment]
I renamed the method to `writeUsingOutputFormat` and rebased to current master.

**Comment by rmetzger** [comment]
I'll merge the PR.

**Pull request closed without merging**